lib/route-recognizer.js

```diff
@@ -85,32 +85,41 @@ EpsilonSegment.prototype = {
   generate: function() { return ""; }
 };
 
-function parse(route, names, types) {
+function parse(route, names, specificity) {
   // normalize route as not starting with a "/". Recognition will
   // also normalize.
   if (route.charAt(0) === "/") { route = route.substr(1); }
 
   var segments = route.split("/"), results = [];
 
+  // We take advantage of the fact that route specificity
+  // mirrors the way we write numbers as strings. For more on
+  // how this works, refer to:
+  // https://github.com/tildeio/route-recognizer/pull/46
```

Comment:
I opted out of writing a paragraph about how this works. Lemme know if you think I should do otherwise.